components/merkledb/src/proof_map_index/proof.rs

```diff
@@ -260,16 +262,25 @@ impl<K, V> OptionalEntry<K, V> {
 /// (e.g., ordering of `proof`; see [`check()`]) only if these invariants are checked
 /// during proof verification.
 ///
-/// [`get_proof()`]: struct.ProofMapIndex.html#method.get_proof
-/// [`get_multiproof()`]: struct.ProofMapIndex.html#method.get_multiproof
+/// [`get_proof()`]: type.ProofMapIndex.html#method.get_proof
+/// [`get_multiproof()`]: type.ProofMapIndex.html#method.get_multiproof
 /// [`check()`]: #method.check
 /// [`ProofPath`]: struct.ProofPath.html
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
-pub struct MapProof<K, V> {
+pub struct MapProof<K, V, KeyMode = Hashed> {
     entries: Vec<OptionalEntry<K, V>>,
     proof: Vec<MapProofEntry>,
+    #[serde(skip)]
+    _key_mode: PhantomData<KeyMode>,
 }
 
+/// Variant of [`MapProof`] with raw keys, in other words with keys that mapped
+/// directly to [`ProofPath`].
+///
+/// [`MapProof`]: struct.MapProof.html
+/// [`ProofPath`]: struct.ProofPath.html
+pub type RawMapProof<K, V> = MapProof<K, V, Raw>;
```

Comment:
Nit / subjective: Is this type alias really necessary? IMO, `MapProof<K, V, Raw>` is just as readable.